{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "11b32aa9-843a-4fc9-884d-2ba214f293c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# This is code to parse a h5ad file into a single delimited cell type\n",
    "# for generation of a Brooklyn plot. The output files are a new delimited\n",
    "# h5ad file of a single cell type and a csv file of genes with their \n",
    "# chromosome locations and mean expression from that cell type.\n",
    "# Code written by Arun Patil and edited by Marc Halushka\n",
    "# CC BY 2022"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "862412b7-a4f7-4ae8-9d09-f633675811be",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import scipy as sp\n",
    "import scanpy as sc\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import glob\n",
    "import os, fnmatch\n",
    "import requests\n",
    "import io\n",
    "import seaborn as sns\n",
    "import scipy\n",
    "import pybiomart\n",
    "sc.settings.verbosity = 3\n",
    "sc.settings.set_figure_params(dpi = 120, color_map = 'RdBu_r')\n",
    "from scipy import stats\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0c95cfc6-4a5b-43cd-9831-0150bcdf6747",
   "metadata": {},
   "source": [
    "# Import the original .h5ad file here. Chose the appropriate file location."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "7035f4d1-3a51-43e6-a32a-e40fee6ac211",
   "metadata": {},
   "outputs": [],
   "source": [
    "orig_h5ad = sc.read_h5ad('subset_Cells_DCM_normalized_humanLV_112122.h5ad')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ff64632f-e7ac-4267-95b3-79726587a1f9",
   "metadata": {},
   "source": [
    "# Obtain the observations (obs) list of the h5ad file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e365a6b4-241b-460c-9892-ba10e863c493",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "AnnData object with n_obs × n_vars = 69150 × 33234\n",
       "    obs: 'Sample', 'Patient', 'Region_x', 'Primary.Genetic.Diagnosis', 'n_genes', 'n_counts', 'percent_mito', 'percent_ribo', 'scrublet_score_z', 'scrublet_score_log', 'solo_score', 'cell_states', 'Assigned', 'ethnicity_ontology_term_id', 'disease_ontology_term_id', 'cell_type_ontology_term_id', 'sex_ontology_term_id', 'assay_ontology_term_id', 'organism_ontology_term_id', 'is_primary_data', 'tissue_ontology_term_id', 'development_stage_ontology_term_id', 'cell_type', 'assay', 'disease', 'organism', 'sex', 'tissue', 'ethnicity', 'development_stage'\n",
       "    var: 'vst.mean', 'vst.variance', 'vst.variance.expected', 'vst.variance.standardized', 'vst.variable', 'feature_biotype', 'feature_is_filtered', 'feature_name', 'feature_reference'\n",
       "    uns: 'X_normalization', 'cell_states_colors', 'cell_type_ontology_term_id_colors', 'layer_descriptions', 'schema_version', 'title'\n",
       "    obsm: 'X_pca', 'X_umap'"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orig_h5ad"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "e1b27663-9f54-4e14-9a76-ac00b7b8e98f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Region_x</th>\n",
       "      <th>Primary.Genetic.Diagnosis</th>\n",
       "      <th>n_genes</th>\n",
       "      <th>n_counts</th>\n",
       "      <th>percent_mito</th>\n",
       "      <th>percent_ribo</th>\n",
       "      <th>scrublet_score_z</th>\n",
       "      <th>scrublet_score_log</th>\n",
       "      <th>...</th>\n",
       "      <th>tissue_ontology_term_id</th>\n",
       "      <th>development_stage_ontology_term_id</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>assay</th>\n",
       "      <th>disease</th>\n",
       "      <th>organism</th>\n",
       "      <th>sex</th>\n",
       "      <th>tissue</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>development_stage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3350</th>\n",
       "      <td>ED_DT4_LV0_premrna</td>\n",
       "      <td>DT4</td>\n",
       "      <td>LV</td>\n",
       "      <td>TTN</td>\n",
       "      <td>2536</td>\n",
       "      <td>2920.292236</td>\n",
       "      <td>0.000278</td>\n",
       "      <td>0.001113</td>\n",
       "      <td>0.066042</td>\n",
       "      <td>0.020795</td>\n",
       "      <td>...</td>\n",
       "      <td>UBERON:0002084</td>\n",
       "      <td>HsapDv:0000241</td>\n",
       "      <td>cardiac muscle cell</td>\n",
       "      <td>10x 3' v3</td>\n",
       "      <td>dilated cardiomyopathy</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>male</td>\n",
       "      <td>heart left ventricle</td>\n",
       "      <td>unknown</td>\n",
       "      <td>seventh decade human stage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3351</th>\n",
       "      <td>ED_DT4_LV0_premrna</td>\n",
       "      <td>DT4</td>\n",
       "      <td>LV</td>\n",
       "      <td>TTN</td>\n",
       "      <td>887</td>\n",
       "      <td>1920.013428</td>\n",
       "      <td>0.001704</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.183444</td>\n",
       "      <td>0.001330</td>\n",
       "      <td>...</td>\n",
       "      <td>UBERON:0002084</td>\n",
       "      <td>HsapDv:0000241</td>\n",
       "      <td>cardiac muscle cell</td>\n",
       "      <td>10x 3' v3</td>\n",
       "      <td>dilated cardiomyopathy</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>male</td>\n",
       "      <td>heart left ventricle</td>\n",
       "      <td>unknown</td>\n",
       "      <td>seventh decade human stage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3352</th>\n",
       "      <td>ED_DT4_LV0_premrna</td>\n",
       "      <td>DT4</td>\n",
       "      <td>LV</td>\n",
       "      <td>TTN</td>\n",
       "      <td>2568</td>\n",
       "      <td>3276.649414</td>\n",
       "      <td>0.003481</td>\n",
       "      <td>0.001266</td>\n",
       "      <td>0.090403</td>\n",
       "      <td>0.083051</td>\n",
       "      <td>...</td>\n",
       "      <td>UBERON:0002084</td>\n",
       "      <td>HsapDv:0000241</td>\n",
       "      <td>cardiac muscle cell</td>\n",
       "      <td>10x 3' v3</td>\n",
       "      <td>dilated cardiomyopathy</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>male</td>\n",
       "      <td>heart left ventricle</td>\n",
       "      <td>unknown</td>\n",
       "      <td>seventh decade human stage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3353</th>\n",
       "      <td>ED_DT4_LV0_premrna</td>\n",
       "      <td>DT4</td>\n",
       "      <td>LV</td>\n",
       "      <td>TTN</td>\n",
       "      <td>1643</td>\n",
       "      <td>2521.693604</td>\n",
       "      <td>0.000748</td>\n",
       "      <td>0.000499</td>\n",
       "      <td>0.067423</td>\n",
       "      <td>0.004540</td>\n",
       "      <td>...</td>\n",
       "      <td>UBERON:0002084</td>\n",
       "      <td>HsapDv:0000241</td>\n",
       "      <td>cardiac muscle cell</td>\n",
       "      <td>10x 3' v3</td>\n",
       "      <td>dilated cardiomyopathy</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>male</td>\n",
       "      <td>heart left ventricle</td>\n",
       "      <td>unknown</td>\n",
       "      <td>seventh decade human stage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3354</th>\n",
       "      <td>ED_DT4_LV0_premrna</td>\n",
       "      <td>DT4</td>\n",
       "      <td>LV</td>\n",
       "      <td>TTN</td>\n",
       "      <td>1832</td>\n",
       "      <td>3079.880615</td>\n",
       "      <td>0.000304</td>\n",
       "      <td>0.000304</td>\n",
       "      <td>0.094364</td>\n",
       "      <td>0.007266</td>\n",
       "      <td>...</td>\n",
       "      <td>UBERON:0002084</td>\n",
       "      <td>HsapDv:0000241</td>\n",
       "      <td>cardiac muscle cell</td>\n",
       "      <td>10x 3' v3</td>\n",
       "      <td>dilated cardiomyopathy</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>male</td>\n",
       "      <td>heart left ventricle</td>\n",
       "      <td>unknown</td>\n",
       "      <td>seventh decade human stage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>310662</th>\n",
       "      <td>IC_H04_LV0_premrna</td>\n",
       "      <td>IC_H04</td>\n",
       "      <td>LV</td>\n",
       "      <td>PVneg</td>\n",
       "      <td>1894</td>\n",
       "      <td>2627.427490</td>\n",
       "      <td>0.003090</td>\n",
       "      <td>0.000813</td>\n",
       "      <td>0.009396</td>\n",
       "      <td>0.004142</td>\n",
       "      <td>...</td>\n",
       "      <td>UBERON:0002084</td>\n",
       "      <td>HsapDv:0000241</td>\n",
       "      <td>cardiac muscle cell</td>\n",
       "      <td>10x 3' v3</td>\n",
       "      <td>dilated cardiomyopathy</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>female</td>\n",
       "      <td>heart left ventricle</td>\n",
       "      <td>unknown</td>\n",
       "      <td>seventh decade human stage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>310663</th>\n",
       "      <td>IC_H04_LV0_premrna</td>\n",
       "      <td>IC_H04</td>\n",
       "      <td>LV</td>\n",
       "      <td>PVneg</td>\n",
       "      <td>2941</td>\n",
       "      <td>3399.208252</td>\n",
       "      <td>0.001667</td>\n",
       "      <td>0.001334</td>\n",
       "      <td>0.028765</td>\n",
       "      <td>0.005743</td>\n",
       "      <td>...</td>\n",
       "      <td>UBERON:0002084</td>\n",
       "      <td>HsapDv:0000241</td>\n",
       "      <td>cardiac muscle cell</td>\n",
       "      <td>10x 3' v3</td>\n",
       "      <td>dilated cardiomyopathy</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>female</td>\n",
       "      <td>heart left ventricle</td>\n",
       "      <td>unknown</td>\n",
       "      <td>seventh decade human stage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>310664</th>\n",
       "      <td>IC_H04_LV0_premrna</td>\n",
       "      <td>IC_H04</td>\n",
       "      <td>LV</td>\n",
       "      <td>PVneg</td>\n",
       "      <td>2479</td>\n",
       "      <td>3142.647461</td>\n",
       "      <td>0.000858</td>\n",
       "      <td>0.000572</td>\n",
       "      <td>0.020787</td>\n",
       "      <td>0.004737</td>\n",
       "      <td>...</td>\n",
       "      <td>UBERON:0002084</td>\n",
       "      <td>HsapDv:0000241</td>\n",
       "      <td>cardiac muscle cell</td>\n",
       "      <td>10x 3' v3</td>\n",
       "      <td>dilated cardiomyopathy</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>female</td>\n",
       "      <td>heart left ventricle</td>\n",
       "      <td>unknown</td>\n",
       "      <td>seventh decade human stage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>310665</th>\n",
       "      <td>IC_H04_LV0_premrna</td>\n",
       "      <td>IC_H04</td>\n",
       "      <td>LV</td>\n",
       "      <td>PVneg</td>\n",
       "      <td>2645</td>\n",
       "      <td>3106.231201</td>\n",
       "      <td>0.001000</td>\n",
       "      <td>0.002000</td>\n",
       "      <td>0.020787</td>\n",
       "      <td>0.003347</td>\n",
       "      <td>...</td>\n",
       "      <td>UBERON:0002084</td>\n",
       "      <td>HsapDv:0000241</td>\n",
       "      <td>cardiac muscle cell</td>\n",
       "      <td>10x 3' v3</td>\n",
       "      <td>dilated cardiomyopathy</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>female</td>\n",
       "      <td>heart left ventricle</td>\n",
       "      <td>unknown</td>\n",
       "      <td>seventh decade human stage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>310666</th>\n",
       "      <td>IC_H04_LV0_premrna</td>\n",
       "      <td>IC_H04</td>\n",
       "      <td>LV</td>\n",
       "      <td>PVneg</td>\n",
       "      <td>474</td>\n",
       "      <td>1415.674561</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001773</td>\n",
       "      <td>0.063112</td>\n",
       "      <td>0.013699</td>\n",
       "      <td>...</td>\n",
       "      <td>UBERON:0002084</td>\n",
       "      <td>HsapDv:0000241</td>\n",
       "      <td>cardiac muscle cell</td>\n",
       "      <td>10x 3' v3</td>\n",
       "      <td>dilated cardiomyopathy</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>female</td>\n",
       "      <td>heart left ventricle</td>\n",
       "      <td>unknown</td>\n",
       "      <td>seventh decade human stage</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>69150 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Sample Patient Region_x Primary.Genetic.Diagnosis  \\\n",
       "3350    ED_DT4_LV0_premrna     DT4       LV                       TTN   \n",
       "3351    ED_DT4_LV0_premrna     DT4       LV                       TTN   \n",
       "3352    ED_DT4_LV0_premrna     DT4       LV                       TTN   \n",
       "3353    ED_DT4_LV0_premrna     DT4       LV                       TTN   \n",
       "3354    ED_DT4_LV0_premrna     DT4       LV                       TTN   \n",
       "...                    ...     ...      ...                       ...   \n",
       "310662  IC_H04_LV0_premrna  IC_H04       LV                     PVneg   \n",
       "310663  IC_H04_LV0_premrna  IC_H04       LV                     PVneg   \n",
       "310664  IC_H04_LV0_premrna  IC_H04       LV                     PVneg   \n",
       "310665  IC_H04_LV0_premrna  IC_H04       LV                     PVneg   \n",
       "310666  IC_H04_LV0_premrna  IC_H04       LV                     PVneg   \n",
       "\n",
       "        n_genes     n_counts  percent_mito  percent_ribo  scrublet_score_z  \\\n",
       "3350       2536  2920.292236      0.000278      0.001113          0.066042   \n",
       "3351        887  1920.013428      0.001704      0.000000          0.183444   \n",
       "3352       2568  3276.649414      0.003481      0.001266          0.090403   \n",
       "3353       1643  2521.693604      0.000748      0.000499          0.067423   \n",
       "3354       1832  3079.880615      0.000304      0.000304          0.094364   \n",
       "...         ...          ...           ...           ...               ...   \n",
       "310662     1894  2627.427490      0.003090      0.000813          0.009396   \n",
       "310663     2941  3399.208252      0.001667      0.001334          0.028765   \n",
       "310664     2479  3142.647461      0.000858      0.000572          0.020787   \n",
       "310665     2645  3106.231201      0.001000      0.002000          0.020787   \n",
       "310666      474  1415.674561      0.000000      0.001773          0.063112   \n",
       "\n",
       "        scrublet_score_log  ...  tissue_ontology_term_id  \\\n",
       "3350              0.020795  ...           UBERON:0002084   \n",
       "3351              0.001330  ...           UBERON:0002084   \n",
       "3352              0.083051  ...           UBERON:0002084   \n",
       "3353              0.004540  ...           UBERON:0002084   \n",
       "3354              0.007266  ...           UBERON:0002084   \n",
       "...                    ...  ...                      ...   \n",
       "310662            0.004142  ...           UBERON:0002084   \n",
       "310663            0.005743  ...           UBERON:0002084   \n",
       "310664            0.004737  ...           UBERON:0002084   \n",
       "310665            0.003347  ...           UBERON:0002084   \n",
       "310666            0.013699  ...           UBERON:0002084   \n",
       "\n",
       "       development_stage_ontology_term_id            cell_type      assay  \\\n",
       "3350                       HsapDv:0000241  cardiac muscle cell  10x 3' v3   \n",
       "3351                       HsapDv:0000241  cardiac muscle cell  10x 3' v3   \n",
       "3352                       HsapDv:0000241  cardiac muscle cell  10x 3' v3   \n",
       "3353                       HsapDv:0000241  cardiac muscle cell  10x 3' v3   \n",
       "3354                       HsapDv:0000241  cardiac muscle cell  10x 3' v3   \n",
       "...                                   ...                  ...        ...   \n",
       "310662                     HsapDv:0000241  cardiac muscle cell  10x 3' v3   \n",
       "310663                     HsapDv:0000241  cardiac muscle cell  10x 3' v3   \n",
       "310664                     HsapDv:0000241  cardiac muscle cell  10x 3' v3   \n",
       "310665                     HsapDv:0000241  cardiac muscle cell  10x 3' v3   \n",
       "310666                     HsapDv:0000241  cardiac muscle cell  10x 3' v3   \n",
       "\n",
       "                       disease      organism     sex                tissue  \\\n",
       "3350    dilated cardiomyopathy  Homo sapiens    male  heart left ventricle   \n",
       "3351    dilated cardiomyopathy  Homo sapiens    male  heart left ventricle   \n",
       "3352    dilated cardiomyopathy  Homo sapiens    male  heart left ventricle   \n",
       "3353    dilated cardiomyopathy  Homo sapiens    male  heart left ventricle   \n",
       "3354    dilated cardiomyopathy  Homo sapiens    male  heart left ventricle   \n",
       "...                        ...           ...     ...                   ...   \n",
       "310662  dilated cardiomyopathy  Homo sapiens  female  heart left ventricle   \n",
       "310663  dilated cardiomyopathy  Homo sapiens  female  heart left ventricle   \n",
       "310664  dilated cardiomyopathy  Homo sapiens  female  heart left ventricle   \n",
       "310665  dilated cardiomyopathy  Homo sapiens  female  heart left ventricle   \n",
       "310666  dilated cardiomyopathy  Homo sapiens  female  heart left ventricle   \n",
       "\n",
       "       ethnicity           development_stage  \n",
       "3350     unknown  seventh decade human stage  \n",
       "3351     unknown  seventh decade human stage  \n",
       "3352     unknown  seventh decade human stage  \n",
       "3353     unknown  seventh decade human stage  \n",
       "3354     unknown  seventh decade human stage  \n",
       "...          ...                         ...  \n",
       "310662   unknown  seventh decade human stage  \n",
       "310663   unknown  seventh decade human stage  \n",
       "310664   unknown  seventh decade human stage  \n",
       "310665   unknown  seventh decade human stage  \n",
       "310666   unknown  seventh decade human stage  \n",
       "\n",
       "[69150 rows x 30 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orig_h5ad.obs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "768c2292-60b4-457a-aec5-27b34d73f3d9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Sample', 'Patient', 'Region_x', 'Primary.Genetic.Diagnosis', 'n_genes',\n",
       "       'n_counts', 'percent_mito', 'percent_ribo', 'scrublet_score_z',\n",
       "       'scrublet_score_log', 'solo_score', 'cell_states', 'Assigned',\n",
       "       'ethnicity_ontology_term_id', 'disease_ontology_term_id',\n",
       "       'cell_type_ontology_term_id', 'sex_ontology_term_id',\n",
       "       'assay_ontology_term_id', 'organism_ontology_term_id',\n",
       "       'is_primary_data', 'tissue_ontology_term_id',\n",
       "       'development_stage_ontology_term_id', 'cell_type', 'assay', 'disease',\n",
       "       'organism', 'sex', 'tissue', 'ethnicity', 'development_stage'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orig_h5ad.obs.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "f2ccebe6-5de5-47ee-901f-23e0d1197132",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>vst.mean</th>\n",
       "      <th>vst.variance</th>\n",
       "      <th>vst.variance.expected</th>\n",
       "      <th>vst.variance.standardized</th>\n",
       "      <th>vst.variable</th>\n",
       "      <th>feature_biotype</th>\n",
       "      <th>feature_is_filtered</th>\n",
       "      <th>feature_name</th>\n",
       "      <th>feature_reference</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_ids</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000243485</th>\n",
       "      <td>0.000071</td>\n",
       "      <td>0.000071</td>\n",
       "      <td>0.000074</td>\n",
       "      <td>0.958714</td>\n",
       "      <td>0</td>\n",
       "      <td>gene</td>\n",
       "      <td>False</td>\n",
       "      <td>MIR1302-2HG</td>\n",
       "      <td>NCBITaxon:9606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000237613</th>\n",
       "      <td>0.000003</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>0.997981</td>\n",
       "      <td>0</td>\n",
       "      <td>gene</td>\n",
       "      <td>False</td>\n",
       "      <td>FAM138A</td>\n",
       "      <td>NCBITaxon:9606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000186092</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>gene</td>\n",
       "      <td>False</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>NCBITaxon:9606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000238009</th>\n",
       "      <td>0.031093</td>\n",
       "      <td>0.032638</td>\n",
       "      <td>0.035845</td>\n",
       "      <td>0.910519</td>\n",
       "      <td>0</td>\n",
       "      <td>gene</td>\n",
       "      <td>False</td>\n",
       "      <td>RP11-34P13.7</td>\n",
       "      <td>NCBITaxon:9606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000239945</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>gene</td>\n",
       "      <td>False</td>\n",
       "      <td>RP11-34P13.8</td>\n",
       "      <td>NCBITaxon:9606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000277856</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>gene</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000277856</td>\n",
       "      <td>NCBITaxon:9606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000275063</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>gene</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000275063</td>\n",
       "      <td>NCBITaxon:9606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000271254</th>\n",
       "      <td>0.005992</td>\n",
       "      <td>0.006174</td>\n",
       "      <td>0.006542</td>\n",
       "      <td>0.943884</td>\n",
       "      <td>0</td>\n",
       "      <td>gene</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000271254</td>\n",
       "      <td>NCBITaxon:9606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000277475</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>gene</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000277475</td>\n",
       "      <td>NCBITaxon:9606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000268674</th>\n",
       "      <td>0.000003</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>0.997981</td>\n",
       "      <td>0</td>\n",
       "      <td>gene</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000268674</td>\n",
       "      <td>NCBITaxon:9606</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>33234 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 vst.mean  vst.variance  vst.variance.expected  \\\n",
       "gene_ids                                                         \n",
       "ENSG00000243485  0.000071      0.000071               0.000074   \n",
       "ENSG00000237613  0.000003      0.000003               0.000003   \n",
       "ENSG00000186092  0.000000      0.000000               0.000000   \n",
       "ENSG00000238009  0.031093      0.032638               0.035845   \n",
       "ENSG00000239945  0.000000      0.000000               0.000000   \n",
       "...                   ...           ...                    ...   \n",
       "ENSG00000277856  0.000000      0.000000               0.000000   \n",
       "ENSG00000275063  0.000000      0.000000               0.000000   \n",
       "ENSG00000271254  0.005992      0.006174               0.006542   \n",
       "ENSG00000277475  0.000000      0.000000               0.000000   \n",
       "ENSG00000268674  0.000003      0.000003               0.000003   \n",
       "\n",
       "                 vst.variance.standardized  vst.variable feature_biotype  \\\n",
       "gene_ids                                                                   \n",
       "ENSG00000243485                   0.958714             0            gene   \n",
       "ENSG00000237613                   0.997981             0            gene   \n",
       "ENSG00000186092                   0.000000             0            gene   \n",
       "ENSG00000238009                   0.910519             0            gene   \n",
       "ENSG00000239945                   0.000000             0            gene   \n",
       "...                                    ...           ...             ...   \n",
       "ENSG00000277856                   0.000000             0            gene   \n",
       "ENSG00000275063                   0.000000             0            gene   \n",
       "ENSG00000271254                   0.943884             0            gene   \n",
       "ENSG00000277475                   0.000000             0            gene   \n",
       "ENSG00000268674                   0.997981             0            gene   \n",
       "\n",
       "                 feature_is_filtered     feature_name feature_reference  \n",
       "gene_ids                                                                 \n",
       "ENSG00000243485                False      MIR1302-2HG    NCBITaxon:9606  \n",
       "ENSG00000237613                False          FAM138A    NCBITaxon:9606  \n",
       "ENSG00000186092                False            OR4F5    NCBITaxon:9606  \n",
       "ENSG00000238009                False     RP11-34P13.7    NCBITaxon:9606  \n",
       "ENSG00000239945                False     RP11-34P13.8    NCBITaxon:9606  \n",
       "...                              ...              ...               ...  \n",
       "ENSG00000277856                False  ENSG00000277856    NCBITaxon:9606  \n",
       "ENSG00000275063                False  ENSG00000275063    NCBITaxon:9606  \n",
       "ENSG00000271254                False  ENSG00000271254    NCBITaxon:9606  \n",
       "ENSG00000277475                False  ENSG00000277475    NCBITaxon:9606  \n",
       "ENSG00000268674                False  ENSG00000268674    NCBITaxon:9606  \n",
       "\n",
       "[33234 rows x 9 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orig_h5ad.var"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "306680b7-901e-4f69-bdcb-5921c393fd90",
   "metadata": {},
   "source": [
    "# Determine all of the cell types available in the .h5ad file. Determining the options in any obs can be done as similar as for cell_type."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "382fb299-6c9d-4b44-84f4-9688493eff62",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3350      cardiac muscle cell\n",
       "3351      cardiac muscle cell\n",
       "3352      cardiac muscle cell\n",
       "3353      cardiac muscle cell\n",
       "3354      cardiac muscle cell\n",
       "                 ...         \n",
       "310662    cardiac muscle cell\n",
       "310663    cardiac muscle cell\n",
       "310664    cardiac muscle cell\n",
       "310665    cardiac muscle cell\n",
       "310666    cardiac muscle cell\n",
       "Name: cell_type, Length: 69150, dtype: category\n",
       "Categories (1, object): ['cardiac muscle cell']"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orig_h5ad.obs[\"cell_type\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c19bfd01-c3b4-418d-8f9c-44f18b25003e",
   "metadata": {},
   "source": [
    "# Obtain a single cell type from the h5ad file from a single 'obs'. Or use the option below to add a single cell type bases on multiple delimiters."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "6a8383d9-6c20-4a87-bfe3-86b2117285b9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3350        TTN\n",
       "3351        TTN\n",
       "3352        TTN\n",
       "3353        TTN\n",
       "3354        TTN\n",
       "          ...  \n",
       "310662    PVneg\n",
       "310663    PVneg\n",
       "310664    PVneg\n",
       "310665    PVneg\n",
       "310666    PVneg\n",
       "Name: Primary.Genetic.Diagnosis, Length: 69150, dtype: category\n",
       "Categories (11, object): ['DES', 'DSP', 'FKTN', 'FLNC', ..., 'RBM20', 'TNNC1', 'TNNT2', 'TTN']"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orig_h5ad.obs[\"Primary.Genetic.Diagnosis\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "42ac98db-7da6-44cb-bd01-90eae757e385",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3350      dilated cardiomyopathy\n",
       "3351      dilated cardiomyopathy\n",
       "3352      dilated cardiomyopathy\n",
       "3353      dilated cardiomyopathy\n",
       "3354      dilated cardiomyopathy\n",
       "                   ...          \n",
       "310662    dilated cardiomyopathy\n",
       "310663    dilated cardiomyopathy\n",
       "310664    dilated cardiomyopathy\n",
       "310665    dilated cardiomyopathy\n",
       "310666    dilated cardiomyopathy\n",
       "Name: disease, Length: 69150, dtype: category\n",
       "Categories (1, object): ['dilated cardiomyopathy']"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orig_h5ad.obs[\"disease\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "dec80107-3892-418c-996e-4e654ce2dde5",
   "metadata": {},
   "outputs": [],
   "source": [
    "onecell_h5ad = orig_h5ad[orig_h5ad.obs['cell_type'] == 'cardiac muscle cell']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "cb155aea-b8ac-4c59-aef8-8ae3bdbab36c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# If you have multiple delimiters for your cell type of interest, remove the '#' below and use this code\n",
    "# to futher subset based on different obs codes of orig_h5ad above.  Check the column names of obs to use\n",
    "# this correctly.  Column names will vary between h5ad sets so please edit accordingly.\n",
    "# example1 - onecell_h5ad = orig_h5ad[(orig_h5ad.obs['disease'] == 'dilated cardiomyopathy') & (orig_h5ad.obs['Region_x'] == 'LV')]\n",
    "# example2 - onecell_h5ad = orig_h5ad[(orig_h5ad.obs['disease'] == 'dilated cardiomyopathy') & (orig_h5ad.obs['Region_x'] == 'LV') & (orig_h5ad.obs['Primary.Genetic.Diagnosis'] == 'PVneg')]\n",
    "\n",
    "onecell_h5ad = orig_h5ad[(orig_h5ad.obs['disease'] == 'dilated cardiomyopathy') & (orig_h5ad.obs['Region_x'] == 'LV') & (orig_h5ad.obs['Primary.Genetic.Diagnosis'] == 'TTN')]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cf0d4f39-d695-45f1-9253-e75783f34c57",
   "metadata": {},
   "source": [
    "# Establish the shape of the subset matrix and compare it to the original matrix. The onecell_h5ad have fewer rows."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "0a6e544f-f13e-4472-83d4-9f949802836b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(69150, 33234)"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orig_h5ad.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "ce4ba67e-49f9-4c05-aa12-fd413cf90cd2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(17965, 33234)"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "onecell_h5ad.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "88c8b006-ffe1-4933-a61e-463978e81ebc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>vst.mean</th>\n",
       "      <th>vst.variance</th>\n",
       "      <th>vst.variance.expected</th>\n",
       "      <th>vst.variance.standardized</th>\n",
       "      <th>vst.variable</th>\n",
       "      <th>feature_biotype</th>\n",
       "      <th>feature_is_filtered</th>\n",
       "      <th>feature_name</th>\n",
       "      <th>feature_reference</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_ids</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000243485</th>\n",
       "      <td>0.000071</td>\n",
       "      <td>0.000071</td>\n",
       "      <td>0.000074</td>\n",
       "      <td>0.958714</td>\n",
       "      <td>0</td>\n",
       "      <td>gene</td>\n",
       "      <td>False</td>\n",
       "      <td>MIR1302-2HG</td>\n",
       "      <td>NCBITaxon:9606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000237613</th>\n",
       "      <td>0.000003</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>0.997981</td>\n",
       "      <td>0</td>\n",
       "      <td>gene</td>\n",
       "      <td>False</td>\n",
       "      <td>FAM138A</td>\n",
       "      <td>NCBITaxon:9606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000186092</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>gene</td>\n",
       "      <td>False</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>NCBITaxon:9606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000238009</th>\n",
       "      <td>0.031093</td>\n",
       "      <td>0.032638</td>\n",
       "      <td>0.035845</td>\n",
       "      <td>0.910519</td>\n",
       "      <td>0</td>\n",
       "      <td>gene</td>\n",
       "      <td>False</td>\n",
       "      <td>RP11-34P13.7</td>\n",
       "      <td>NCBITaxon:9606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000239945</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>gene</td>\n",
       "      <td>False</td>\n",
       "      <td>RP11-34P13.8</td>\n",
       "      <td>NCBITaxon:9606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000277856</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>gene</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000277856</td>\n",
       "      <td>NCBITaxon:9606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000275063</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>gene</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000275063</td>\n",
       "      <td>NCBITaxon:9606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000271254</th>\n",
       "      <td>0.005992</td>\n",
       "      <td>0.006174</td>\n",
       "      <td>0.006542</td>\n",
       "      <td>0.943884</td>\n",
       "      <td>0</td>\n",
       "      <td>gene</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000271254</td>\n",
       "      <td>NCBITaxon:9606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000277475</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>gene</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000277475</td>\n",
       "      <td>NCBITaxon:9606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000268674</th>\n",
       "      <td>0.000003</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>0.997981</td>\n",
       "      <td>0</td>\n",
       "      <td>gene</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000268674</td>\n",
       "      <td>NCBITaxon:9606</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>33234 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 vst.mean  vst.variance  vst.variance.expected  \\\n",
       "gene_ids                                                         \n",
       "ENSG00000243485  0.000071      0.000071               0.000074   \n",
       "ENSG00000237613  0.000003      0.000003               0.000003   \n",
       "ENSG00000186092  0.000000      0.000000               0.000000   \n",
       "ENSG00000238009  0.031093      0.032638               0.035845   \n",
       "ENSG00000239945  0.000000      0.000000               0.000000   \n",
       "...                   ...           ...                    ...   \n",
       "ENSG00000277856  0.000000      0.000000               0.000000   \n",
       "ENSG00000275063  0.000000      0.000000               0.000000   \n",
       "ENSG00000271254  0.005992      0.006174               0.006542   \n",
       "ENSG00000277475  0.000000      0.000000               0.000000   \n",
       "ENSG00000268674  0.000003      0.000003               0.000003   \n",
       "\n",
       "                 vst.variance.standardized  vst.variable feature_biotype  \\\n",
       "gene_ids                                                                   \n",
       "ENSG00000243485                   0.958714             0            gene   \n",
       "ENSG00000237613                   0.997981             0            gene   \n",
       "ENSG00000186092                   0.000000             0            gene   \n",
       "ENSG00000238009                   0.910519             0            gene   \n",
       "ENSG00000239945                   0.000000             0            gene   \n",
       "...                                    ...           ...             ...   \n",
       "ENSG00000277856                   0.000000             0            gene   \n",
       "ENSG00000275063                   0.000000             0            gene   \n",
       "ENSG00000271254                   0.943884             0            gene   \n",
       "ENSG00000277475                   0.000000             0            gene   \n",
       "ENSG00000268674                   0.997981             0            gene   \n",
       "\n",
       "                 feature_is_filtered     feature_name feature_reference  \n",
       "gene_ids                                                                 \n",
       "ENSG00000243485                False      MIR1302-2HG    NCBITaxon:9606  \n",
       "ENSG00000237613                False          FAM138A    NCBITaxon:9606  \n",
       "ENSG00000186092                False            OR4F5    NCBITaxon:9606  \n",
       "ENSG00000238009                False     RP11-34P13.7    NCBITaxon:9606  \n",
       "ENSG00000239945                False     RP11-34P13.8    NCBITaxon:9606  \n",
       "...                              ...              ...               ...  \n",
       "ENSG00000277856                False  ENSG00000277856    NCBITaxon:9606  \n",
       "ENSG00000275063                False  ENSG00000275063    NCBITaxon:9606  \n",
       "ENSG00000271254                False  ENSG00000271254    NCBITaxon:9606  \n",
       "ENSG00000277475                False  ENSG00000277475    NCBITaxon:9606  \n",
       "ENSG00000268674                False  ENSG00000268674    NCBITaxon:9606  \n",
       "\n",
       "[33234 rows x 9 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "onecell_h5ad.var"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bea7d1a9-f5d4-450e-97e6-d235812f9fe3",
   "metadata": {},
   "source": [
    "# Create the subsetted h5ad file with the specific cell type - delimited by cell type and any other qualifiers needed."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "7e8b7e79-892d-457e-982f-21e5f8ce1dbd",
   "metadata": {},
   "outputs": [],
   "source": [
    "onecell_h5ad.write_h5ad(\"subset_seidman_TTN.h5ad\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "34916900-3c46-486a-8ba3-cd359e3f96b1",
   "metadata": {},
   "source": [
    "# Dataset is a variable obtained from the biomart annotation needed for the Brooklyn plot."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "bf68fff2-8e65-464c-8139-2c5902f8961b",
   "metadata": {},
   "outputs": [],
   "source": [
    "dataset = sc.queries.biomart_annotations(\n",
    "        \"hsapiens\",\n",
    "        [\"ensembl_gene_id\", \"start_position\", \"end_position\", \"chromosome_name\", \"hgnc_symbol\", \"band\"],\n",
    "    ).set_index(\"ensembl_gene_id\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "a3b1c842-667b-4c1f-9126-1d4ee146d592",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "      <th>chromosome_name</th>\n",
       "      <th>hgnc_symbol</th>\n",
       "      <th>band</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000210049</th>\n",
       "      <td>577</td>\n",
       "      <td>647</td>\n",
       "      <td>MT</td>\n",
       "      <td>MT-TF</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000211459</th>\n",
       "      <td>648</td>\n",
       "      <td>1601</td>\n",
       "      <td>MT</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000210077</th>\n",
       "      <td>1602</td>\n",
       "      <td>1670</td>\n",
       "      <td>MT</td>\n",
       "      <td>MT-TV</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000210082</th>\n",
       "      <td>1671</td>\n",
       "      <td>3229</td>\n",
       "      <td>MT</td>\n",
       "      <td>MT-RNR2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000209082</th>\n",
       "      <td>3230</td>\n",
       "      <td>3304</td>\n",
       "      <td>MT</td>\n",
       "      <td>MT-TL1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000162437</th>\n",
       "      <td>64745075</td>\n",
       "      <td>64833232</td>\n",
       "      <td>1</td>\n",
       "      <td>RAVER2</td>\n",
       "      <td>p31.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000122432</th>\n",
       "      <td>84506300</td>\n",
       "      <td>84567379</td>\n",
       "      <td>1</td>\n",
       "      <td>SPATA1</td>\n",
       "      <td>p22.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000284882</th>\n",
       "      <td>84574114</td>\n",
       "      <td>84583620</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>p22.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000289881</th>\n",
       "      <td>84614068</td>\n",
       "      <td>84621061</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>p22.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000285325</th>\n",
       "      <td>84785427</td>\n",
       "      <td>84786714</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>p22.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>69305 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 start_position  end_position chromosome_name hgnc_symbol  \\\n",
       "ensembl_gene_id                                                             \n",
       "ENSG00000210049             577           647              MT       MT-TF   \n",
       "ENSG00000211459             648          1601              MT     MT-RNR1   \n",
       "ENSG00000210077            1602          1670              MT       MT-TV   \n",
       "ENSG00000210082            1671          3229              MT     MT-RNR2   \n",
       "ENSG00000209082            3230          3304              MT      MT-TL1   \n",
       "...                         ...           ...             ...         ...   \n",
       "ENSG00000162437        64745075      64833232               1      RAVER2   \n",
       "ENSG00000122432        84506300      84567379               1      SPATA1   \n",
       "ENSG00000284882        84574114      84583620               1         NaN   \n",
       "ENSG00000289881        84614068      84621061               1         NaN   \n",
       "ENSG00000285325        84785427      84786714               1         NaN   \n",
       "\n",
       "                  band  \n",
       "ensembl_gene_id         \n",
       "ENSG00000210049    NaN  \n",
       "ENSG00000211459    NaN  \n",
       "ENSG00000210077    NaN  \n",
       "ENSG00000210082    NaN  \n",
       "ENSG00000209082    NaN  \n",
       "...                ...  \n",
       "ENSG00000162437  p31.3  \n",
       "ENSG00000122432  p22.3  \n",
       "ENSG00000284882  p22.3  \n",
       "ENSG00000289881  p22.3  \n",
       "ENSG00000285325  p22.3  \n",
       "\n",
       "[69305 rows x 5 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataset"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "68471052-55c4-4115-b81a-d271f2c4cdce",
   "metadata": {},
   "source": [
    "# Converting raw expression count data and converting to a numpy array."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "8a6b3f13-1efa-415b-ab90-2da0b699f817",
   "metadata": {},
   "outputs": [],
   "source": [
    "data = onecell_h5ad.X.toarray()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e1762785-9636-4c5e-8c99-ef47708c601c",
   "metadata": {},
   "source": [
    "#  This demonstrates what the array values look like."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "8369465b-4bc7-4fa0-bd91-fc9a4e32f19f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[0., 0., 0., ..., 0., 0., 0.],\n",
       "       [0., 0., 0., ..., 0., 0., 0.],\n",
       "       [0., 0., 0., ..., 0., 0., 0.],\n",
       "       [0., 0., 0., ..., 0., 0., 0.],\n",
       "       [0., 0., 0., ..., 0., 0., 0.]], dtype=float32)"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data[:5]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9c0e7cb5-e85c-46ee-a57c-c3d1c7731ac3",
   "metadata": {},
   "source": [
    "# This indicates how many genes are in the array (length), which can be important for generating the Brooklyn plot.  Ideally, the original h5ad array does not limit the number of genes from the original sequencing."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "4fbfbd34-2004-4276-b4dc-2e716b5a37ca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['ENSG00000243485', 'ENSG00000237613', 'ENSG00000186092',\n",
       "       'ENSG00000238009', 'ENSG00000239945', 'ENSG00000239906',\n",
       "       'ENSG00000241599', 'ENSG00000236601', 'ENSG00000284733',\n",
       "       'ENSG00000235146',\n",
       "       ...\n",
       "       'ENSG00000277196', 'ENSG00000277630', 'ENSG00000278384',\n",
       "       'ENSG00000278633', 'ENSG00000276345', 'ENSG00000277856',\n",
       "       'ENSG00000275063', 'ENSG00000271254', 'ENSG00000277475',\n",
       "       'ENSG00000268674'],\n",
       "      dtype='object', name='gene_ids', length=33234)"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "onecell_h5ad.var_names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "1f50cdd0-890a-44cc-9013-11dcc2678b89",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<69150x33234 sparse matrix of type '<class 'numpy.float32'>'\n",
       "\twith 161117875 stored elements in Compressed Sparse Row format>"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orig_h5ad.X"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a20be494-7528-4291-98ab-4f361426263a",
   "metadata": {},
   "source": [
    "# This code generates mean values for each gene (ENSG ID), from raw expression data, and appends this to gene annotations from biomart."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "547ef7ad-d7a3-43ad-b402-335532c27108",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>xMean</th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "      <th>chromosome_name</th>\n",
       "      <th>hgnc_symbol</th>\n",
       "      <th>band</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000000003</th>\n",
       "      <td>0.004342</td>\n",
       "      <td>100627108</td>\n",
       "      <td>100639991</td>\n",
       "      <td>X</td>\n",
       "      <td>TSPAN6</td>\n",
       "      <td>q22.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000005</th>\n",
       "      <td>0.001058</td>\n",
       "      <td>100584936</td>\n",
       "      <td>100599885</td>\n",
       "      <td>X</td>\n",
       "      <td>TNMD</td>\n",
       "      <td>q22.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000419</th>\n",
       "      <td>0.360610</td>\n",
       "      <td>50934867</td>\n",
       "      <td>50959140</td>\n",
       "      <td>20</td>\n",
       "      <td>DPM1</td>\n",
       "      <td>q13.13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000457</th>\n",
       "      <td>0.165547</td>\n",
       "      <td>169849631</td>\n",
       "      <td>169894267</td>\n",
       "      <td>1</td>\n",
       "      <td>SCYL3</td>\n",
       "      <td>q24.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000460</th>\n",
       "      <td>0.155693</td>\n",
       "      <td>169662007</td>\n",
       "      <td>169854080</td>\n",
       "      <td>1</td>\n",
       "      <td>C1orf112</td>\n",
       "      <td>q24.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000285492</th>\n",
       "      <td>0.002394</td>\n",
       "      <td>159051674</td>\n",
       "      <td>159121506</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>q25.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000285505</th>\n",
       "      <td>0.054773</td>\n",
       "      <td>41956879</td>\n",
       "      <td>41994232</td>\n",
       "      <td>19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>q13.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000285508</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>10413520</td>\n",
       "      <td>10431922</td>\n",
       "      <td>20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>p12.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000285509</th>\n",
       "      <td>0.012970</td>\n",
       "      <td>121024125</td>\n",
       "      <td>121113108</td>\n",
       "      <td>11</td>\n",
       "      <td>TBCEL-TECTA</td>\n",
       "      <td>q23.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000285513</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>116820645</td>\n",
       "      <td>116821541</td>\n",
       "      <td>11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>q23.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>33147 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    xMean  start_position  end_position chromosome_name  \\\n",
       "ENSG00000000003  0.004342       100627108     100639991               X   \n",
       "ENSG00000000005  0.001058       100584936     100599885               X   \n",
       "ENSG00000000419  0.360610        50934867      50959140              20   \n",
       "ENSG00000000457  0.165547       169849631     169894267               1   \n",
       "ENSG00000000460  0.155693       169662007     169854080               1   \n",
       "...                   ...             ...           ...             ...   \n",
       "ENSG00000285492  0.002394       159051674     159121506               6   \n",
       "ENSG00000285505  0.054773        41956879      41994232              19   \n",
       "ENSG00000285508  0.000000        10413520      10431922              20   \n",
       "ENSG00000285509  0.012970       121024125     121113108              11   \n",
       "ENSG00000285513  0.000000       116820645     116821541              11   \n",
       "\n",
       "                 hgnc_symbol    band  \n",
       "ENSG00000000003       TSPAN6   q22.1  \n",
       "ENSG00000000005         TNMD   q22.1  \n",
       "ENSG00000000419         DPM1  q13.13  \n",
       "ENSG00000000457        SCYL3   q24.2  \n",
       "ENSG00000000460     C1orf112   q24.2  \n",
       "...                      ...     ...  \n",
       "ENSG00000285492          NaN   q25.3  \n",
       "ENSG00000285505          NaN   q13.2  \n",
       "ENSG00000285508          NaN   p12.2  \n",
       "ENSG00000285509  TBCEL-TECTA   q23.3  \n",
       "ENSG00000285513          NaN   q23.3  \n",
       "\n",
       "[33147 rows x 6 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "xmen = onecell_h5ad.raw.X.mean(0)\n",
    "type(xmen)\n",
    "xmendf = pd.DataFrame(xmen.T, columns = ['xMean'], index = onecell_h5ad.var_names)\n",
    "xmendf\n",
    "df_cd = pd.merge(xmendf, dataset, left_index=True, right_index=True)\n",
    "df_cd"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "448d6c85-597d-4f2c-9682-b4c6d5c906ec",
   "metadata": {},
   "source": [
    "# This resets the index and adds a name for the first column (ENSG ID)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "091eb8db-047e-4e77-a2aa-e20612fc93f8",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_cd_export = df_cd.reset_index().rename(columns={'index': 'gene_ids'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "69b76931-976f-4376-95ec-6613ba21f0a2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_ids</th>\n",
       "      <th>xMean</th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "      <th>chromosome_name</th>\n",
       "      <th>hgnc_symbol</th>\n",
       "      <th>band</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>0.004342</td>\n",
       "      <td>100627108</td>\n",
       "      <td>100639991</td>\n",
       "      <td>X</td>\n",
       "      <td>TSPAN6</td>\n",
       "      <td>q22.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000005</td>\n",
       "      <td>0.001058</td>\n",
       "      <td>100584936</td>\n",
       "      <td>100599885</td>\n",
       "      <td>X</td>\n",
       "      <td>TNMD</td>\n",
       "      <td>q22.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000000419</td>\n",
       "      <td>0.360610</td>\n",
       "      <td>50934867</td>\n",
       "      <td>50959140</td>\n",
       "      <td>20</td>\n",
       "      <td>DPM1</td>\n",
       "      <td>q13.13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000000457</td>\n",
       "      <td>0.165547</td>\n",
       "      <td>169849631</td>\n",
       "      <td>169894267</td>\n",
       "      <td>1</td>\n",
       "      <td>SCYL3</td>\n",
       "      <td>q24.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000000460</td>\n",
       "      <td>0.155693</td>\n",
       "      <td>169662007</td>\n",
       "      <td>169854080</td>\n",
       "      <td>1</td>\n",
       "      <td>C1orf112</td>\n",
       "      <td>q24.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33142</th>\n",
       "      <td>ENSG00000285492</td>\n",
       "      <td>0.002394</td>\n",
       "      <td>159051674</td>\n",
       "      <td>159121506</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>q25.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33143</th>\n",
       "      <td>ENSG00000285505</td>\n",
       "      <td>0.054773</td>\n",
       "      <td>41956879</td>\n",
       "      <td>41994232</td>\n",
       "      <td>19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>q13.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33144</th>\n",
       "      <td>ENSG00000285508</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>10413520</td>\n",
       "      <td>10431922</td>\n",
       "      <td>20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>p12.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33145</th>\n",
       "      <td>ENSG00000285509</td>\n",
       "      <td>0.012970</td>\n",
       "      <td>121024125</td>\n",
       "      <td>121113108</td>\n",
       "      <td>11</td>\n",
       "      <td>TBCEL-TECTA</td>\n",
       "      <td>q23.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33146</th>\n",
       "      <td>ENSG00000285513</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>116820645</td>\n",
       "      <td>116821541</td>\n",
       "      <td>11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>q23.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>33147 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              gene_ids     xMean  start_position  end_position  \\\n",
       "0      ENSG00000000003  0.004342       100627108     100639991   \n",
       "1      ENSG00000000005  0.001058       100584936     100599885   \n",
       "2      ENSG00000000419  0.360610        50934867      50959140   \n",
       "3      ENSG00000000457  0.165547       169849631     169894267   \n",
       "4      ENSG00000000460  0.155693       169662007     169854080   \n",
       "...                ...       ...             ...           ...   \n",
       "33142  ENSG00000285492  0.002394       159051674     159121506   \n",
       "33143  ENSG00000285505  0.054773        41956879      41994232   \n",
       "33144  ENSG00000285508  0.000000        10413520      10431922   \n",
       "33145  ENSG00000285509  0.012970       121024125     121113108   \n",
       "33146  ENSG00000285513  0.000000       116820645     116821541   \n",
       "\n",
       "      chromosome_name  hgnc_symbol    band  \n",
       "0                   X       TSPAN6   q22.1  \n",
       "1                   X         TNMD   q22.1  \n",
       "2                  20         DPM1  q13.13  \n",
       "3                   1        SCYL3   q24.2  \n",
       "4                   1     C1orf112   q24.2  \n",
       "...               ...          ...     ...  \n",
       "33142               6          NaN   q25.3  \n",
       "33143              19          NaN   q13.2  \n",
       "33144              20          NaN   p12.2  \n",
       "33145              11  TBCEL-TECTA   q23.3  \n",
       "33146              11          NaN   q23.3  \n",
       "\n",
       "[33147 rows x 7 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cd_export"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7cfca239-201c-4e7e-be19-ce64376d39c6",
   "metadata": {},
   "source": [
    "# This command exports a csv file that can be used to pick (automated or manually), genes spread across the whole genome for Brooklyn plots."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "15ecacbb-c24c-42e6-8bc2-802cf29ae287",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_cd_export.to_csv(\"seidmanttn_var_biomart.csv\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f762ba64-e639-49a3-ad5e-6fe9d38f0e79",
   "metadata": {},
   "source": [
    "# This part of the code removes the top 3,500 genes by raw xMean, then sorts by chromosome position and generates gene lists of all 3,500 genes and 350 interspersed genes to generate the Brooklyn plot needed for the next python step."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "963f9054-8358-4036-afee-d5b2c705b92b",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_cd_export2 = df_cd_export.sort_values(by = ['xMean'],ascending = False).reset_index(drop = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "e3674de8-cad6-4222-a957-3519f23cec41",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_cd_export_top = df_cd_export2.iloc[:3500,:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "0d6afa8b-60d0-4f14-a91c-3e29ebf0c1e0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_ids</th>\n",
       "      <th>xMean</th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "      <th>chromosome_name</th>\n",
       "      <th>hgnc_symbol</th>\n",
       "      <th>band</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000251562</td>\n",
       "      <td>1618.217773</td>\n",
       "      <td>65497688</td>\n",
       "      <td>65506516</td>\n",
       "      <td>11</td>\n",
       "      <td>MALAT1</td>\n",
       "      <td>q13.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000198626</td>\n",
       "      <td>146.410553</td>\n",
       "      <td>237042184</td>\n",
       "      <td>237833988</td>\n",
       "      <td>1</td>\n",
       "      <td>RYR2</td>\n",
       "      <td>q43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000155657</td>\n",
       "      <td>100.666641</td>\n",
       "      <td>178525989</td>\n",
       "      <td>178830802</td>\n",
       "      <td>2</td>\n",
       "      <td>TTN</td>\n",
       "      <td>q31.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000245532</td>\n",
       "      <td>39.976116</td>\n",
       "      <td>65422774</td>\n",
       "      <td>65445540</td>\n",
       "      <td>11</td>\n",
       "      <td>NEAT1</td>\n",
       "      <td>q13.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000183023</td>\n",
       "      <td>38.942490</td>\n",
       "      <td>40097270</td>\n",
       "      <td>40611053</td>\n",
       "      <td>2</td>\n",
       "      <td>SLC8A1</td>\n",
       "      <td>p22.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3495</th>\n",
       "      <td>ENSG00000231672</td>\n",
       "      <td>0.338896</td>\n",
       "      <td>217284019</td>\n",
       "      <td>217756593</td>\n",
       "      <td>2</td>\n",
       "      <td>DIRC3</td>\n",
       "      <td>q35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3496</th>\n",
       "      <td>ENSG00000059758</td>\n",
       "      <td>0.338786</td>\n",
       "      <td>96278261</td>\n",
       "      <td>96400480</td>\n",
       "      <td>12</td>\n",
       "      <td>CDK17</td>\n",
       "      <td>q23.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3497</th>\n",
       "      <td>ENSG00000134375</td>\n",
       "      <td>0.338454</td>\n",
       "      <td>201955503</td>\n",
       "      <td>201970664</td>\n",
       "      <td>1</td>\n",
       "      <td>TIMM17A</td>\n",
       "      <td>q32.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3498</th>\n",
       "      <td>ENSG00000070018</td>\n",
       "      <td>0.338398</td>\n",
       "      <td>12116025</td>\n",
       "      <td>12267044</td>\n",
       "      <td>12</td>\n",
       "      <td>LRP6</td>\n",
       "      <td>p13.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3499</th>\n",
       "      <td>ENSG00000110811</td>\n",
       "      <td>0.338174</td>\n",
       "      <td>6828407</td>\n",
       "      <td>6839847</td>\n",
       "      <td>12</td>\n",
       "      <td>P3H3</td>\n",
       "      <td>p13.31</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3500 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             gene_ids        xMean  start_position  end_position  \\\n",
       "0     ENSG00000251562  1618.217773        65497688      65506516   \n",
       "1     ENSG00000198626   146.410553       237042184     237833988   \n",
       "2     ENSG00000155657   100.666641       178525989     178830802   \n",
       "3     ENSG00000245532    39.976116        65422774      65445540   \n",
       "4     ENSG00000183023    38.942490        40097270      40611053   \n",
       "...               ...          ...             ...           ...   \n",
       "3495  ENSG00000231672     0.338896       217284019     217756593   \n",
       "3496  ENSG00000059758     0.338786        96278261      96400480   \n",
       "3497  ENSG00000134375     0.338454       201955503     201970664   \n",
       "3498  ENSG00000070018     0.338398        12116025      12267044   \n",
       "3499  ENSG00000110811     0.338174         6828407       6839847   \n",
       "\n",
       "     chromosome_name hgnc_symbol    band  \n",
       "0                 11      MALAT1   q13.1  \n",
       "1                  1        RYR2     q43  \n",
       "2                  2         TTN   q31.2  \n",
       "3                 11       NEAT1   q13.1  \n",
       "4                  2      SLC8A1   p22.1  \n",
       "...              ...         ...     ...  \n",
       "3495               2       DIRC3     q35  \n",
       "3496              12       CDK17   q23.1  \n",
       "3497               1     TIMM17A   q32.1  \n",
       "3498              12        LRP6   p13.2  \n",
       "3499              12        P3H3  p13.31  \n",
       "\n",
       "[3500 rows x 7 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cd_export_top"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "60ea7263-e92d-4bfd-8d25-5a55b9bf1170",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_cd_export_genelist = df_cd_export_top.sort_values(by = ['chromosome_name','end_position']).reset_index(drop = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "1d6fc5c5-95ab-4ec0-b5eb-15882c4ed59d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_ids</th>\n",
       "      <th>xMean</th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "      <th>chromosome_name</th>\n",
       "      <th>hgnc_symbol</th>\n",
       "      <th>band</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000187642</td>\n",
       "      <td>0.430749</td>\n",
       "      <td>975198</td>\n",
       "      <td>982117</td>\n",
       "      <td>1</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>p36.33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000221978</td>\n",
       "      <td>0.356993</td>\n",
       "      <td>1385711</td>\n",
       "      <td>1399335</td>\n",
       "      <td>1</td>\n",
       "      <td>CCNL2</td>\n",
       "      <td>p36.33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000189409</td>\n",
       "      <td>0.413938</td>\n",
       "      <td>1632163</td>\n",
       "      <td>1635263</td>\n",
       "      <td>1</td>\n",
       "      <td>MMP23B</td>\n",
       "      <td>p36.33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000078369</td>\n",
       "      <td>0.475395</td>\n",
       "      <td>1785285</td>\n",
       "      <td>1892292</td>\n",
       "      <td>1</td>\n",
       "      <td>GNB1</td>\n",
       "      <td>p36.33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000142611</td>\n",
       "      <td>0.866854</td>\n",
       "      <td>3069168</td>\n",
       "      <td>3438621</td>\n",
       "      <td>1</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>p36.32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3495</th>\n",
       "      <td>ENSG00000114374</td>\n",
       "      <td>0.857443</td>\n",
       "      <td>12537650</td>\n",
       "      <td>12860839</td>\n",
       "      <td>Y</td>\n",
       "      <td>USP9Y</td>\n",
       "      <td>q11.221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3496</th>\n",
       "      <td>ENSG00000183878</td>\n",
       "      <td>1.383058</td>\n",
       "      <td>13234577</td>\n",
       "      <td>13480673</td>\n",
       "      <td>Y</td>\n",
       "      <td>UTY</td>\n",
       "      <td>q11.221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3497</th>\n",
       "      <td>ENSG00000176728</td>\n",
       "      <td>3.638888</td>\n",
       "      <td>18772706</td>\n",
       "      <td>19077416</td>\n",
       "      <td>Y</td>\n",
       "      <td>TTTY14</td>\n",
       "      <td>q11.222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3498</th>\n",
       "      <td>ENSG00000229236</td>\n",
       "      <td>0.921071</td>\n",
       "      <td>20464916</td>\n",
       "      <td>20575519</td>\n",
       "      <td>Y</td>\n",
       "      <td>TTTY10</td>\n",
       "      <td>q11.223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3499</th>\n",
       "      <td>ENSG00000198692</td>\n",
       "      <td>0.371463</td>\n",
       "      <td>20575776</td>\n",
       "      <td>20593154</td>\n",
       "      <td>Y</td>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>q11.223</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3500 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             gene_ids     xMean  start_position  end_position chromosome_name  \\\n",
       "0     ENSG00000187642  0.430749          975198        982117               1   \n",
       "1     ENSG00000221978  0.356993         1385711       1399335               1   \n",
       "2     ENSG00000189409  0.413938         1632163       1635263               1   \n",
       "3     ENSG00000078369  0.475395         1785285       1892292               1   \n",
       "4     ENSG00000142611  0.866854         3069168       3438621               1   \n",
       "...               ...       ...             ...           ...             ...   \n",
       "3495  ENSG00000114374  0.857443        12537650      12860839               Y   \n",
       "3496  ENSG00000183878  1.383058        13234577      13480673               Y   \n",
       "3497  ENSG00000176728  3.638888        18772706      19077416               Y   \n",
       "3498  ENSG00000229236  0.921071        20464916      20575519               Y   \n",
       "3499  ENSG00000198692  0.371463        20575776      20593154               Y   \n",
       "\n",
       "     hgnc_symbol     band  \n",
       "0          PERM1   p36.33  \n",
       "1          CCNL2   p36.33  \n",
       "2         MMP23B   p36.33  \n",
       "3           GNB1   p36.33  \n",
       "4         PRDM16   p36.32  \n",
       "...          ...      ...  \n",
       "3495       USP9Y  q11.221  \n",
       "3496         UTY  q11.221  \n",
       "3497      TTTY14  q11.222  \n",
       "3498      TTTY10  q11.223  \n",
       "3499      EIF1AY  q11.223  \n",
       "\n",
       "[3500 rows x 7 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cd_export_genelist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "cba4040c-9439-40fb-aa97-9d39a290dcfd",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_cd_export_againstlist = df_cd_export_genelist.iloc[:3500,:1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "4e9b9776-c4aa-4b3e-bad4-4f824d292a05",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_ids</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000187642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000221978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000189409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000078369</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000142611</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3495</th>\n",
       "      <td>ENSG00000114374</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3496</th>\n",
       "      <td>ENSG00000183878</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3497</th>\n",
       "      <td>ENSG00000176728</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3498</th>\n",
       "      <td>ENSG00000229236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3499</th>\n",
       "      <td>ENSG00000198692</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3500 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             gene_ids\n",
       "0     ENSG00000187642\n",
       "1     ENSG00000221978\n",
       "2     ENSG00000189409\n",
       "3     ENSG00000078369\n",
       "4     ENSG00000142611\n",
       "...               ...\n",
       "3495  ENSG00000114374\n",
       "3496  ENSG00000183878\n",
       "3497  ENSG00000176728\n",
       "3498  ENSG00000229236\n",
       "3499  ENSG00000198692\n",
       "\n",
       "[3500 rows x 1 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cd_export_againstlist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "c55fd368-5c9e-40e7-ac08-982d29ad48cd",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_cd_export_againstlist.to_csv(\"againstlist.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "47127bd7-f67a-480d-9f57-857df783fe78",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_cd_export_genelist_parse = df_cd_export_genelist.iloc[1::10, :]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "3416da15-b235-46a7-b2fa-f05a81a92115",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_cd_export_towrite = df_cd_export_genelist_parse.iloc[:350,:1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "74fa71e3-dea6-4596-ba0e-30b6e287412e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_ids</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000221978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ENSG00000180758</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>ENSG00000175206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>ENSG00000037637</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>ENSG00000169641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3451</th>\n",
       "      <td>ENSG00000232593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3461</th>\n",
       "      <td>ENSG00000229807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3471</th>\n",
       "      <td>ENSG00000166432</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3481</th>\n",
       "      <td>ENSG00000101972</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3491</th>\n",
       "      <td>ENSG00000102125</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>350 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             gene_ids\n",
       "1     ENSG00000221978\n",
       "11    ENSG00000180758\n",
       "21    ENSG00000175206\n",
       "31    ENSG00000037637\n",
       "41    ENSG00000169641\n",
       "...               ...\n",
       "3451  ENSG00000232593\n",
       "3461  ENSG00000229807\n",
       "3471  ENSG00000166432\n",
       "3481  ENSG00000101972\n",
       "3491  ENSG00000102125\n",
       "\n",
       "[350 rows x 1 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cd_export_towrite"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "4beafbf3-7f49-4032-b203-641420097ba6",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_cd_export_towrite.to_csv(\"genelist.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9be3df55-ed77-4ccd-a15c-4ba731493c74",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
